contracts/ETHMixer.sol

```diff
@@ -22,18 +22,20 @@ contract ETHMixer is Mixer {
   ) Mixer(_verifier, _denomination, _merkleTreeHeight, _operator) public {
   }
 
-  function _processDeposit() internal {
+  function _processDeposit() internal nonReentrant {
```

Comment:
This function does not modify blockchain state at all, so “nonReentrant” modifier on it looks weird.  Consider moving modifier to the calling function that actually ought to be protected.

Reply:
Agree